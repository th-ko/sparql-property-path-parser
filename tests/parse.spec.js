var {parse} = require('../index')

describe('property path parser', () => {
 it('can parse simple sequences', () => {
    const ast = parse('propOne/propTwo/propThree')
    expect(ast).toMatchSnapshot()
 })
 it('can parse complex paths', () => {
    const ast = parse('altOnePropOne{1,}/altOnePropTwo+|altTwoPropOne*/propThree{3}')
    expect(ast).toMatchSnapshot()
 })
 it('can parse path with spaces', () => {
    const ast = parse('propOne / ( altPropOne | altPropTwo)/propThree*')
    expect(ast).toMatchSnapshot()
 })
 it('can inline sequences that are backeted', () => {
    const ast = parse('propOne / ( propTwo / propThree)/(propFour/propFive)')
    expect(ast).toMatchSnapshot()
 })
})