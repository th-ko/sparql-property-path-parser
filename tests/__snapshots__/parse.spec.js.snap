// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`property path parser can inline sequences that are backeted 1`] = `
Object {
  "paths": Array [
    Object {
      "path": "propOne",
    },
    Object {
      "paths": Array [
        Object {
          "path": "propTwo",
        },
        Object {
          "path": "propThree",
        },
      ],
      "type": "sequence",
    },
    Object {
      "paths": Array [
        Object {
          "path": "propFour",
        },
        Object {
          "path": "propFive",
        },
      ],
      "type": "sequence",
    },
  ],
  "type": "sequence",
}
`;

exports[`property path parser can parse complex paths 1`] = `
Object {
  "paths": Array [
    Object {
      "paths": Array [
        Object {
          "modifier": Object {
            "max": Infinity,
            "min": 1,
          },
          "path": "altOnePropOne",
        },
        Object {
          "modifier": Object {
            "max": Infinity,
            "min": 1,
          },
          "path": "altOnePropTwo",
        },
      ],
      "type": "sequence",
    },
    Object {
      "paths": Array [
        Object {
          "modifier": Object {
            "max": Infinity,
            "min": 0,
          },
          "path": "altTwoPropOne",
        },
        Object {
          "modifier": Object {
            "max": 3,
            "min": 3,
          },
          "path": "propThree",
        },
      ],
      "type": "sequence",
    },
  ],
  "type": "alternative",
}
`;

exports[`property path parser can parse path with spaces 1`] = `
Object {
  "paths": Array [
    Object {
      "path": "propOne",
    },
    Object {
      "paths": Array [
        Object {
          "path": "altPropOne",
        },
        Object {
          "path": "altPropTwo",
        },
      ],
      "type": "alternative",
    },
    Object {
      "modifier": Object {
        "max": Infinity,
        "min": 0,
      },
      "path": Object {
        "path": "propThree",
      },
    },
  ],
  "type": "sequence",
}
`;

exports[`property path parser can parse simple sequences 1`] = `
Object {
  "paths": Array [
    Object {
      "path": "propOne",
    },
    Object {
      "path": "propTwo",
    },
    Object {
      "path": "propThree",
    }
  ],
  "type": "sequence",
}
`;
